README.md
# Getting Started with Create React App

This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).

## Available Scripts

In the project directory, you can run:

### `yarn install`
`EXPORT READ_TOKEN=TOKEN` 
To do before to run yarn install in order to get the private repo. You can find the token in bitwarden

### `yarn start`

Runs the app in the development mode.\
Open [http://localhost:3000](http://localhost:3000) to view it in the browser.

The page will reload if you make edits.\
You will also see any lint errors in the console.

### `yarn test`

Launches the test runner in the interactive watch mode.\
See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.

### `yarn build`

Builds the app for production to the `build` folder.\
It correctly bundles React in production mode and optimizes the build for the best performance.

The build is minified and the filenames include the hashes.\
Your app is ready to be deployed!

See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.

### `yarn eject`

**Note: this is a one-way operation. Once you `eject`, you can’t go back!**

If you aren’t satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.

Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.

You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.

## Learn More

You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).

To learn React, check out the [React documentation](https://reactjs.org/).

### Code Splitting

This section has moved here: [https://facebook.github.io/create-react-app/docs/code-splitting](https://facebook.github.io/create-react-app/docs/code-splitting)

### Analyzing the Bundle Size

This section has moved here: [https://facebook.github.io/create-react-app/docs/analyzing-the-bundle-size](https://facebook.github.io/create-react-app/docs/analyzing-the-bundle-size)

### Making a Progressive Web App

This section has moved here: [https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app](https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app)

### Advanced Configuration

This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)

### Deployment

This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)

### `yarn build` fails to minify

This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [4.3.0] (29-12-2022)

- back end : 4.3.0
- front end core : 0.8.3

## [4.0.0] - (19-09-2022)
- back end : 3.8.0
- front end core : 0.8.0

## [3.9.1] - (05-10-2022)
- back end : 3.7.1
- front end core : 0.7.1

## [3.9.0] - (05-09-2022)
- back end : 3.7.0
- front end core : 0.7.0

## [3.8.0] (08-08-2022)

- back end : 3.6.0
- front end core : 0.6.0

## [3.7.1] 04-07-2022

- back end : 3.5.1
- front end core : 0.5.6

## [3.7.0] 20-06-2022

- back end : 3.5.1
- front end core : 0.5.5

## [3.6.2] 17-05-2022

- back end : 3.4.2
- front end core : 0.4.2
 
## [3.6.1] 13-05-2022

- back end : 3.4.2
- front end core : 0.4.1

### Fixed 

- edit mode for event form when the event is duplicate

## [3.6.0] 09-05-2022

- back end : 3.4.2
- front end core : 0.4.0

### Added 

-Standardization of the data grid
-Agenda help modal

### Fixed

-Bug up-down on activities data grid


## [3.5.0] 11-04-2022

- back end : 3.4.2
- front end core : 0.3.2

### Added

- Agenda
- Improvement of the presentation of the activities card on participant side.
- Task to export data on a file
- Ordering of the presentation of the activities: the project team get new buttons to change the display priority of one activity.

### Fixed 

- Display of a loader when data load for all grids.
- New participants have a 'Repéré' status by default.

## [3.3.0]

### Added

- (#311) a loader is displayed when datagrid charge datas
- (#136) events can be managed from agenda page (create, update, moved, delete)

### Fixed

- (#323) remove print export option
- display all notes when no filter selected

### Changed

**tech**

- (#304) Internationalisation

## [3.2.0]

### Added

- (#279) activities by category can be moved (up or down) in list

### Fixed

- (#183) Categories card no more hidden when slide left
- (#18) add i18n

## [3.1.0] - 2022-02-22

- (#231) column in events list are changed

### Added

- (#163) notes filters and search components for logbooks
- (#233) Correction on card style for participants

### Fixed

- (#271) Uniformise navbar comportement and styles
- (#286) add two new status for participant (spotted and on break)
- (#159) participant can see his own profile in logbooks

## [3.0.0] - 2022-01-27

### Added

- (#133) Loading while file upload : disable button and change wording
- (#160 + #240) add logbooks and manage notes for participants
- (#162) add and manage (update, delete, mark as view) comments in a note

## [2.4.0] - 2021-12-10

- (#224) add tests for user and documents store
- add lefthook for pre push test

### Added

- (#226) intervener can duplicate a event, with same attributes but must change date
- (#195) add activitiy's long description visible on hover in activitiesList
- (#208) participant can manifest interest for activity without events
- (#148) add columns to activities grid
- (#219) interveners can handle participants interests for activities

**tech**

- isolation tests of `store.documents.fetchDocuments`

### Fixed

- (#226) can change attendance status if accepted or refused, not only destroy
- (#226) change headCellsEventList
- (#217) incorrect gender displayed in the complete profile form
- (#222) participant button in activity's event card is updated

## [2.3.0] - 2021-11-26

### Changed

**tech**

- compatibility update with focale/app-back-end tag:2.3.0

### Fixed

- (#216) profile panel in participant's journey tab displaying 0 instead of
  the birthdate
- (#217) incorrect gender displayed in the complete profile form

## [2.2.1][2.2.1] - 2021-11-12

### Fixed

- event page not displaying only 10 registrations instead of all

## [2.2.0][2.2.0] - 2021-11-12

### Added

- (#172) event without attendances can be deleted
- (#193) (#194) add a attendance feature for add or remove participant in event
- (#125) add a toggle status button in the activity list
- (#125) add a go-to activity button in the activity list
- (#145) (#176) add a document status panel in the participant's journey tab
- (#170) add events creation
- (#175) add a profile panel in the participant's journey tab

### Changed

**tech**

- (#188) tabs navigation keep current tab in memory
- (#178) refactor and optimize the participant's journey tab

### Fixed

- (#191) the username autocomplete is disabled in profile form
- (#146) the profile form not focused on password field
- (#189) the current sorting of the user lists is now more visible
- (#205) participants' sex not correctly displayed
- (#206) incoherent events displayed for each category

### Known issues

- (#204) new participants have no journey status by default

## [2.1.0][2.1.0] - 2021-11-04

### Added

**security**

- (#115) disable inline React JavaScript
- (#115) configure the Content Security Policy
- (#116) make cookies secure

**features**

- (#156) new journey tab in participants' profile, with status management
- (#154) participants can browse activities categories in a new carousel view
- (#154) social network links on the home page
- (#77) interveners can create activities
- (#151) participant can list activities per category

**build**

- fix several build warnings
- add eslint, with no special configuration

### Changed

- add columns related to participants' journey in participant list

### Fixed

- activities categories not properly updated
- birthdate, entry date, and exit date incorrectly saved

## [2.0.0][2.0.0] - unreleased

### Added

- (#147) interveners can list all activities of a catagory

### Changed

- activities have been renamed categories
- (#152) split the store in several files
- (#147) the quick search field is now reusable

### Fixed

- several build warnings

## [1.3.0] - 2021-09-20

### Added

- (#142) add a quick-search field in the users list

src/themes/intervenerTheme.js
import { frFR as coreFrFR } from '@mui/material/locale';
import { createTheme } from '@mui/material/styles';
import { frFR as dataGridFrFR } from '@mui/x-data-grid';

import { commonPalette } from './commonPalette';
import { commonTypo } from './commonPalette';


const { commonPrimary, commonSecondary, calendar, success, warning, white, black, gray, text } = commonPalette;
export const intervenerTheme = createTheme({
  typography:
    commonTypo,
  palette: {
    mode: 'light',
    primary: {
      main: '#4B59C4',
      light:'#8E9AF2',
      dark: '#323B83',
      contrastText: '#FFFFFF',
      accent: '#5F70EE',
      ultralight:'#D4D9FF',
      extralight:'#B0B9FF',
    },
    secondary: {
      main: '#00e4a0',
      light: '#99FFE0',
      dark: '#00bc84',
      contrastText: '#1B1B1B',
      accent: '#3EFCC3',
      extralight: '#C2FFED',
      ultralight: '#D9FFF3'
    },
    commonPrimary,
    commonSecondary,
    calendar,
    success,
    warning,
    gray,
    white,
    black,
    text,
  },
},
dataGridFrFR,
coreFrFR,
);

src/themes/participantTheme.js
import { frFR as coreFrFR } from '@mui/material/locale';
import { createTheme } from '@mui/material/styles';
import { frFR as datagridFrFR} from '@mui/x-data-grid';

import { commonPalette } from './commonPalette';
import { commonTypo } from './commonPalette';


const { commonPrimary, commonSecondary, calendar, success, warning, white, black, gray, text } = commonPalette;
export const participantTheme = createTheme({
  typography:
    commonTypo,
  palette: {
    mode: 'light',
    primary: {
      main: '#00e4a0',
      light: '#99FFE0',
      dark: '#00bc84',
      contrastText: '#1B1B1B',
      accent: '#3EFCC3',
      extralight: '#C2FFED',
      ultralight: '#D9FFF3'
    },
    secondary: {
      main: '#4B59C4',
      light:'#8E9AF2',
      dark: '#323B83',
      contrastText: '#FFFFFF',
      accent: '#5F70EE',
      ultralight:'#D4D9FF',
      extralight:'#B0B9FF',
    },
    commonPrimary,
    commonSecondary,
    calendar,
    success,
    warning,
    white,
    black,
    gray,
    text,
  },
},
datagridFrFR,
coreFrFR
);


src/themes/CustomThemeProvider.js
import React, { useState } from 'react';

import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import PropTypes from 'prop-types';

import getTheme from './base';

export const CustomThemeContext = React.createContext(
  {
    currentTheme: 'participantTheme',
    setTheme: null,
  },
);

const CustomThemeProvider = ({children}) => {
  // Read current theme from localStorage or maybe from an api

  if (localStorage.getItem('appTheme') !== 'participantTheme' 
    || localStorage.getItem('appTheme') !== 'intervenerTheme' ) {
    localStorage.setItem('appTheme', 'participantTheme');
  }

  const currentTheme = localStorage.getItem('appTheme') || 'participantTheme';

  // State to hold the selected theme name
  const [themeName, _setThemeName] = useState(currentTheme);

  // Retrieve the theme object by theme name
  const theme = getTheme(themeName);

  // Wrap _setThemeName to store new theme names in localStorage
  const setThemeName = (name) => {
    localStorage.setItem('appTheme', name);
    _setThemeName(name);
  };

  const contextValue = {
    currentTheme: themeName,
    setTheme: setThemeName,
  };

  return (
    <CustomThemeContext.Provider value={contextValue}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </StyledEngineProvider>
    </CustomThemeContext.Provider>
  );
};

CustomThemeProvider.propTypes = ({
  children: PropTypes.element.isRequired,
});

export default CustomThemeProvider;


src/themes/commonPalette.js
import { deepOrange, green, purple } from '@mui/material/colors';

const commonPalette = {
  white: '#fafafa',
  black: '#000',
  commonPrimary: {
    main: '#4B59C4',
    light:'#8E9AF2',
    dark: '#323B83',
    contrastText: '#FFFFFF',
    accent: '#5F70EE',
    ultralight:'#D4D9FF',
    extralight:'#B0B9FF',
  },
  commonSecondary: {
    main: '#00e4a0',
    light: '#99FFE0',
    dark: '#00bc84',
    contrastText: '#1B1B1B',
    accent: '#3EFCC3',
    extralight: '#C2FFED',
    ultralight: '#D9FFF3'
  },
  success: {
    main: '#8ACD5C',
    light: '#CBFFA8',
    ultralight: '#EAFFDB',
  },
  warning: {
    main: '#FF9800',
    light: '#FFC878',
    ultralight: '#FFEDD4',
  },
  error: {
    main: '#FF6646',
    light: '#FFAA98',
    ultralight: '#FFEFEC',
  },
  gray: {
    main: '#9D9DC6',
    dark: '#7C70A8',
    accent: '#C7BDE3',
    light:'#E1DBF0',
    ultralight:'#D5D5D5',
    extralight:'#EAEAEA',
  },
  text: {
    primary: '#000073',
    secondary: '#8F86B0', // LIGHT in sketch
    disabled: '#B6A7DF',
    // ADDED COLORS
    accent: '#7546D5'
  },
  calendar: {
    myEvent: deepOrange[300],
    meeting: purple.A100,
    other: green.A200,
  }
};

const commonTypo = {
  fontFamily: ['Roboto'].join(','),
  h1: {
    fontSize: 32,
    fontWeight: 700,
  },
  h2: {
    fontSize: 27,
    fontWeight: 700,
  },
  h3: {
    fontSize: 23,
    fontWeight: 700,
  },
  h4: {
    fontSize: 20,
    fontWeight: 500,
  },
  h5: {
    fontSize: 18,
    fontWeight: 500,
  },
  h6: {
    fontSize: 16,
    fontWeight: 500,
    lineHeight: 1.3,
  },
  subtitle1: {
    fontSize: 15,
    fontWeight: 700,
  },
  subtitle2: {
    fontSize: 14,
    fontWeight: 700,
  },
  button: {
    textTransform: 'none',
    fontSize: 17,
    fontWeight: 400,
  },
  caption: {
    fontSize: 13,
    fontWeight: 400,
  },
  overline: {
    fontSize: 17,
    fontWeight: 400,
  },
  body1: {
    fontSize: 16,
    fontWeight: 400,
  },
};

export { commonPalette, commonTypo };
